### PR TITLE
Sign tsh for teleport connect on windows builder

### PR DESCRIPTION
**Pull request by tobiaszheller**
## Background

Currently we are building tsh for windows on linux machine using cross-compiling and there we are using osslsigncode to sign it.
However tsh is also built for teleport connect on windows builder, and there executable is not signed at all. It results in poor UX when using windows hello/fido2 as MFA in Teleport Connect.

## Solution

I have setup new step in windows builder to sign tsh executables. I am using windows signtool to sign it. 

Working pipeline: https://drone.platform.teleport.sh/gravitational/teleport/17291/27/9
Built executable: https://get.gravitational.com/Teleport%20Connect%20Setup-12.0.0-dev.tobiaszheller.7.exe

## Why signtool instead of osslsigncode

I have tried setting up new step for signing tsh using osslsigncode however it resulted in following error:
```
Failed to parse PKCS#12 file: 
digital envelope routines:inner_evp_generic_fetch:unsupported
```
It seems that static build of osslsigncode for windows is using openssl 3.0 which fails on parsing PKCS12 (more details: https://github.com/mtrojnar/osslsigncode/issues/178)

It turned out that windows native tool `signtool` is already installed on windows builder and it can handler PKCS12 without any issues.

## Future todo

- remove osslsigncode from windows builder or  use format supported by openssl 3.0 for signing code.

### CODE REVIEW

**Comment by tobiaszheller** [comment]
@tcsc @alistanis @timothyb89 please take a look

**Review by alistanis — APPROVED** [review]
LGTM and I understand the "why" here but I am far from an expert in this part of our code.

File: dronegen/windows.go [hunk 1]
```diff
@@ -232,7 +233,28 @@ func buildWindowsTshStep(workspace string) step {
...
+			"WINDOWS_SIGNING_CERT": {fromSecret: "WINDOWS_SIGNING_CERT"},
+		},
+		Commands: []string{
+			`$ErrorActionPreference = 'Stop'`,
+			`$Workspace = "` + perBuildWorkspace + `"`,
+			`$TeleportSrc = "$Workspace` + teleportSrc + `"`,
+			`. "$TeleportSrc/build.assets/windows/build.ps1"`,
+			`cd $TeleportSrc`,
+			`([System.Convert]::FromBase64String($ENV:WINDOWS_SIGNING_CERT)) | Set-Content windows-signing-cert.pfx -Encoding Byte`,
+			`& 'C:\Program Files (x86)\Windows Kits\10\App Certification Kit\signtool.exe' sign /f windows-signing-cert.pfx /d Teleport /t http://timestamp.digicert.com /du https://goteleport.com /fd sha256 build\tsh-unsigned.exe`,
```
Comment (tcsc) [inline dronegen/windows.go:255]: This looks fine. Only concern is if the location of `signtool.exe` will change from version to version of VS code.

**Review by tcsc — APPROVED** [review]

File: dronegen/windows.go (hunk 1 above)
Comment (tobiaszheller) [reply dronegen/windows.go:255]: i think it's not gonna change with vs code version but probably with system major update. I don't have though better idea how to search for it.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**